src/screens/signIn/SignInScreen.tsx

```diff
@@ -1,44 +1,95 @@
-import React, {useState} from 'react'
-import {View} from 'react-native'
-import {Button, Input} from 'react-native-elements'
+import React, {FC, useEffect, useRef, useState} from 'react'
+import {
+  ActivityIndicator,
+  Animated,
+  Text,
+  TouchableHighlight,
+  View,
+} from 'react-native'
+import {TextInput} from '../../components/TextInput'
 import {
   sharedUseCreateTokenMutation,
   useCreateTokenMutation,
 } from '../../features/auth'
 import styles from './styles'
 
-export const SignInScreen = () => {
+export const SignInScreen: FC<any> = ({navigation}) => {
   // @TODO: remove placeholder later
   const [username, setUsername] = useState('maksim@example.com')
   const [password, setPassword] = useState('q12345')
-  const [createToken] = useCreateTokenMutation({
-    fixedCacheKey: sharedUseCreateTokenMutation,
-  })
+  const [startAnimatedValue, setAnimatedValue] = useState(0)
+  const animation = useRef(new Animated.Value(startAnimatedValue)).current
+  const [createToken, {error, data, isLoading, isSuccess, isError}] =
+    useCreateTokenMutation({
+      fixedCacheKey: sharedUseCreateTokenMutation,
+    })
+
+  useEffect(() => {
+    if (isError) {
+      Animated.timing(animation, {
+        toValue: 1,
+        duration: 500,
+        useNativeDriver: true,
+      }).start()
+    } else {
+      setAnimatedValue(0)
+    }
+  }, [isError, animation, error])
+
+  useEffect(() => {
+    // @ts-ignore
+    if (error?.status === 'FETCH_ERROR') {
+      navigation.navigate('NoInternetModal', {
+        retry: () => createToken({username, password}),
+      })
+    }
+  }, [error])
 
   const handleSignIn = () => {
     createToken({username, password})
   }
 
   return (
-    <View>
-      <Input
-        placeholder="Username"
-        value={username}
-        onChangeText={setUsername}
-        autoCompleteType={false}
-      />
-      <Input
-        placeholder="Password"
-        secureTextEntry
-        value={password}
-        onChangeText={setPassword}
-        autoCompleteType={false}
-      />
-      <Button
-        style={styles.signInButton}
-        title="Sign in"
-        onPress={handleSignIn}
-      />
+    <View style={styles.root}>
+      <View style={styles.textInputContainer}>
+        <TextInput
+          style={styles.textInput}
```

Comment:
did you add a validation animation, i.e checking that email is in valid format?

Reply:
No, I just handle the response from the server and animate the container with an error if any